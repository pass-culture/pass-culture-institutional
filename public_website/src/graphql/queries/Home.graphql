query Home(
  $filters: ResourceFiltersInput
  $pagination: PaginationArg
  $sort: [String]
) {
  home {
    aboutSection {
      ...ComponentBlockCenteredText
    }
    CTASection {
      ...ComponentBlockPushCta
    }
    socialMediaSection {
      ...ComponentBlockSocialMedia
    }
    latestStudies {
      ...ComponentBlockLatestNews
    }
    eligibilitySection {
      ...ComponentHomeEligibilitySection
    }
    heroSection {
      ...ComponentHomeHeroSection
    }
    recommendationsSection {
      ...ComponentHomeRecommendationsSection
    }
    seo {
      ...ComponentSharedSeo
    }
  }
  resources(filters: $filters, pagination: $pagination, sort: $sort) {
    ...Resource
  }
}
