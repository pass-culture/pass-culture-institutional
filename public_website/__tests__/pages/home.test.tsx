import React from 'react'
import { beforeEach, describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe'

import Home, { getStaticProps } from '../../src/pages'
import { act, render, screen } from '..'

describe('Home page', () => {
  beforeEach(() => {
    process.env = {
      ...process.env,
      ID_TOKEN: 'dummy_token',
      INSTITUTIONAL_API_KEY: 'dummy_key',
      BACKEND_API_URL: 'http://dummy_localhost:5001/',
    }
  })

  it('should pass accessibility tests', async () => {
    const { container } = render(<Home />)

    let a11yResult
    await act(async () => {
      a11yResult = await axe(container)
    })
    expect(a11yResult).toHaveNoViolations()
  })

  it('should render correctly', () => {
    const { container } = render(<Home />)

    expect(container).toMatchSnapshot()
  })

  it('should display the playlist name', async () => {
    const { props } = await getStaticProps()
    render(<Home {...props} />)

    const PLAYLIST_NAME = 'Bons plans du moment'
    expect(screen.queryByText(PLAYLIST_NAME)).toBeTruthy()
  })

  it('should display the playlist', async () => {
    const { props } = await getStaticProps()
    render(<Home {...props} />)

    const FIRST_OFFER = 'Livre 1 avec EAN'
    expect(screen.queryByText(FIRST_OFFER)).toBeTruthy()

    const LAST_OFFER = 'Livre 10 avec EAN'
    expect(screen.queryByText(LAST_OFFER)).toBeTruthy()
  })
})
