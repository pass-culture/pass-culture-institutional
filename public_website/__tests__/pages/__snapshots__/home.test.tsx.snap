// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page > should render correctly 1`] = `
<div>
  <div
    class="sc-gtsqUy dWOEAx"
  >
    <main
      class="sc-bdnyFh cOrhoO"
    >
      <h1
        class="sc-pNWRh djejpM"
      >
        Welcome to
         
        <a
          class="sc-dlniIP LWZKO"
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="sc-jrsKJM eHsNYH"
      >
        Get started by editing
        <code
          class="sc-fujznN SvhJn"
        >
          pages/index.tsx
        </code>
      </p>
      <div
        class="sc-gKAaef cJkFME"
        spaces="2"
      />
      <div
        class="sc-hKFymg jLSsZz"
      >
        <div
          class="sc-eCAqax blQmze"
        >
          <a
            class="sc-jSFipO fkzsOZ"
            href="/about"
          >
            About â†’
          </a>
        </div>
      </div>
      <div
        class="sc-gKAaef cJkFME"
        spaces="2"
      />
    </main>
  </div>
</div>
`;
